File: 2019/hs/app/Main.hs
module Main where

import System.Environment   

import Day1
import Day2
import Day3
import Day4
import Day5
import Day6
import Day7
import Day8
import Day9
import Day10
import Day11
import Day12
import Day13
import Day14
import Day15
import Day16
import Day17
import Day18

days :: [String -> IO ()]
days = [ day1, day2, day3, day4, day5, day6, day7, day8, day9, day10, day11
       , day12, day13, day14, day15, day16, day17, day18
       ]

main :: IO ()
main = do
  argv <- getArgs
  case argv of
    [] -> print "USAGE: aoc19 [DAY]"
    argv' -> do let day = (read $ head argv') :: Int
                input <- readFile $ "../input" ++ show day
                (days !! (day - 1)) input

File: 2019/rs/src/main.rs
mod day1;
mod day19;
mod day2;
mod day3;
mod day4;
mod day5;
mod day6;
mod day7;
mod day9;
mod intcode;

use std::env;

fn stub(_input_path: &str) {}

fn main() {
    let days = [
        day1::day1,
        day2::day2,
        day3::day3,
        day4::day4,
        day5::day5,
        day6::day6,
        day7::day7,
        stub, // day8::day8,
        day9::day9,
        stub, // day10::day10,
        stub, // day11::day11,
        stub, // day12::day12,
        stub, // day13::day13,
        stub, // day14::day14,
        stub, // day15::day15,
        stub, // day16::day16,
        stub, // day17::day17,
        stub, // day18::day18,
        day19::day19,
    ];
    let args: Vec<String> = env::args().collect();
    if args.len() > 1 {
        let day = args[1].parse::<usize>().unwrap();
        print!("Day {}..", day);
        days[day - 1](&format!("../input{}", day)[..]);
        println!(" OK");
    } else {
        for (i, day) in days.iter().enumerate() {
            print!("Day {}..", i + 1);
            day(&format!("../input{}", i + 1)[..]);
            println!(" OK");
        }
    }
}
